library/ctr_drbg.c

```diff
@@ -389,7 +433,30 @@ int mbedtls_ctr_drbg_random_with_add( void *p_rng,
         output_len -= use_len;
     }
 
+#if defined(MBEDTLS_CTR_DRBG_ABI_COMPAT)
     ctr_drbg_update_internal( ctx, add_input );
+#else
+    // avoid double initialization of key for each call
```

Comment:
We could avoid duplicating code by modifying `ctr_drbg_update_internal()` to take a third parameter, a pointer to a `mbedtls_aes_context`. (It would only initialise and free its own internal AES context if this third parameter is `NULL` and would use the one supplied in the parameter otherwise)

What do you think?

Reply:
I think we'll keep this as it is, as `ctr_drbg_update_internal()` is called in 5 different locations and so the number of ifdefs would grow a lot with 2 different prototypes. Also this inlined operation avoids copying bytes around just for the XOR operation, smaller stack usage etc.

Reply:
I don't think we need two different prototypes, so we could save the ifdefs. Nevertheless, I can see the benefits of the current solution.